### DISCUSSION TITLE
Getting status of checks run on a specific commit

### DISCUSSION THREAD

**Discussion by carolinascarneiro** [octokit/.github#46] [Q&A]
Hello,

I'm trying to get the status of the checks made on a certain commit thorugh GitHub Actions just as they're shown on the GitHub Web UI.

I already tried getting the combined status for a specific reference, which is a specific SHA number (https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#get-the-combined-status-for-a-specific-reference) and listing commit statuses [List commit statuses for a reference](https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#list-commit-statuses-for-a-reference) for a specific SHA, and using Octokit for doing it, but both of them return something different than what is shown on the GitHub Repo Web UI:

![image](https://github.com/octokit/.github/assets/49428998/1c3103fe-4d7a-47a8-834a-770332bf1c3a)

How do I get exactly what is on this image? Also requests with GH CLI and Curl did not work. I also tried check runs and check suites and also them returned something completely different.

Thank you so much.

**Comment by gr2m** [comment]
It's been a while since I checked, maybe there is a simpler way, but this is the GraphQL query fragment for a commit I used in the past

```graphql
                oid
                checkSuites(first: 100) {
                  nodes {
                    checkRuns(first: 100) {
                      nodes {
                        name
                        conclusion
                        permalink
                      }
                    }
                  }
                }
                status {
                  state
                  contexts {
                    state
                    targetUrl
                    description
                    context
                  }
                }
```

you then have to reduce the graphql response to the result you want.


with something like this

```js
const statuses = commit.status ? commit.status.contexts : [];
const unsuccessStatuses = statuses.filter(
  (status) => status.state !== "SUCCESS"
);

const unsuccessfulCheckRuns = checkRuns
  .filter(
    (checkRun) =>
      checkRun.conclusion !== "SUCCESS" &&
      checkRun.conclusion !== "NEUTRAL" &&
      checkRun.conclusion !== null
  )
```
